guides/hack/02-source-code-fundamentals/10-comments.md

```diff
@@ -16,14 +16,17 @@ Hack has three comment syntaxes.
 function foo(): void {}
 ```
 
-Multi line comments start with `/*` and end with `*/`. Comments
+Multi-line comments start with `/*` and end with `*/`. Comments
 starting `/**` are also used for documentation.
 
-Single-line comments start with `//` or `#`, and end with a newline.
+Single-line comments start with `//` and end with a newline. 
+
+`#` is not a valid comment character, as it is used to represent an [Enum Class Label](/hack/built-in-types/enum-class-label).
 
 A number of special comments are recognized; they are:
 
-* `// FALLTHROUGH` in [switch statements](../statements/switch.md)
-* `// strict` and `// partial` in [headers](program-structure.md)
+* `// FALLTHROUGH` in [switch statements](/hack/statements/switch)
+* `// strict` and `// partial` in [headers](/hack/source-code-fundamentals/program-structure)
 * `/* HH_FIXME[1234] */` or `/* HH_IGNORE_ERROR[1234] */`, which
   suppresses typechecker error reporting for error 1234.
+* `#!` in `#!/path/to/executable` (e.g. `/usr/bin/hhvm`), which is the shebang line (interpreter directive).
```

Comment:
- this shouldn't be described as a comment
- should use `/usr/bin/env hhvm` instead
- should note that it is only permitted as the very first line

Reply:
I was thinking that it was a special type of comment like the others (because of the # in #!) but the others do all fall into the structure of being within valid comment syntax so you're right it doesn't belong there.